v3/src/components/tool-shelf/tool-shelf.tsx

```diff
@@ -21,14 +44,83 @@ export const ToolShelf = () => {
       })
     },
     toast = useToast(),
-    tableHandler = () => notify('table'),
-    graphHandler = () => notify('graph'),
     mapHandler = () => notify('map'),
-    sliderHandler = () => notify('slider'),
-    calcHandler = () => notify('calculator'),
     textHandler = () => notify('text'),
     pluginsHandler = () => notify('plugins')
 
+  const row = content?.getRowByIndex(0)
+
```

Comment:
I wonder if the handlers for different types of tiles could be stored in the global registry? Basically, could the ToolShelf iterate through the registry without having to know anything about individual types of components itself? That way, adding a new type of component could be nearly completely confined to adding a new entry to the registry?
I'm not sure about all this. Might be worth an architecture conversation that includes at least Kirk and maybe Henry.
Meanwhile, we don't need to prevent this PR in lieu of the conversation.